src/components/accession2/edit/QuantityModal.tsx

```diff
@@ -161,81 +223,134 @@ export default function QuantityModal(props: QuantityModalProps): JSX.Element {
+                required={true}
               />
-            </Grid>
-            <Grid
-              item
-              xs={gridSize()}
-              marginTop={theme.spacing(2)}
-              marginLeft={isMobile ? '0px' : theme.spacing(0.5)}
-              width='100%'
-            >
-              <Dropdown
-                options={preferredUnits}
-                placeholder={strings.SELECT}
-                onChange={onChangeUnit}
-                selectedValue={record.remainingQuantity?.units}
-                fullWidth={true}
-              />
-            </Grid>
+            ) : (
+              <Box display='flex' textAlign='left' alignItems='end'>
+                <Textfield
+                  label={strings.TOTAL_WEIGHT}
+                  id='remainingQuantity'
+                  onChange={(value) => onChangeRemainingQuantity('weight', value as number)}
+                  type='number'
+                  min={0}
+                  disabledCharacters={['-']}
+                  value={
+                    record.remainingQuantity?.units === 'Seeds'
+                      ? record.estimatedWeight?.quantity.toString()
+                      : record.remainingQuantity?.quantity.toString()
+                  }
+                  errorText={totalWeightError}
+                />
+                <Box height={totalWeightError ? '85px' : 'auto'}>
+                  <UnitsSelector
+                    onChange={onChangeRemainingQuantityUnit}
+                    selectedValue={
+                      record.remainingQuantity?.units === 'Seeds'
+                        ? record.estimatedWeight?.units
+                        : record.remainingQuantity?.units
+                    }
+                    className={classes.units}
+                  />
+                </Box>
+              </Box>
+            )}
           </Grid>
-          <Grid item xs={12} textAlign='left'>
-            {record.remainingQuantity?.units &&
-              record.remainingQuantity?.units !== 'Seeds' &&
-              !isUnitInPreferredSystem(
-                record.remainingQuantity.units,
-                userPreferences.preferredWeightSystem as string
-              ) && (
+          {record.remainingQuantity?.units &&
+            record.remainingQuantity?.units !== 'Seeds' &&
+            !isUnitInPreferredSystem(
+              record.remainingQuantity.units,
+              userPreferences.preferredWeightSystem as string
+            ) && (
+              <Grid item xs={12} textAlign='left'>
                 <ConvertedValue
                   quantity={record.remainingQuantity.quantity}
                   unit={record.remainingQuantity.units}
                   showTooltip={true}
                 />
-              )}
-          </Grid>
-          <Grid item xs={12} sx={{ marginTop: theme.spacing(2) }}>
-            {!isCalculatorOpened ? (
+              </Grid>
+            )}
+          <Grid item xs={12}>
```

Comment:
Missed this before, wondering if this has enough standalone functionality worth separating out. Curious to get your thoughts.

Reply:
I ended up leaving it there since we don't use it anywhere else.  I didn't think there would be a lot of value in making it a separate component that's still just local to this file.